[Feature] Support for document uids as Item source

Thogrim1984 opened the pull request:
With the PR, it is possible to drag Actor, Item JournalEntry and JournalEntryPage documents into the item source field. 
With the UUID in the source field, the respective documents can be opened as requested in issue #1271.

Current problems:

- If a UUID is in the source field, the rendering of the sheet takes significantly longer.
- If you drag a JournalEntryPage into the source field, the browser freezes
- If you want to open a (manually entered) UUID of a JournalEntryPage from the source field, the browser also freezes.
Both freezes occur regardless of whether an anchor is specified or not.

I can't find a solution for these problems because unfortunately I have no idea what the cause could be. Maybe someone can take a look at this.

Thogrim1984 commented:
I think the points 

> 
> - If you drag a JournalEntryPage into the source field, the browser freezes
> - If you want to open a (manually entered) UUID of a JournalEntryPage from the source field, the browser also freezes.
> Both freezes occur regardless of whether an anchor is specified or not.

are due to this issue: #1285

taMiF reviewed (changes requested):
I restructured some parts some parts so it matches previous structure better.

I also added
- changing icons based on source type for uuid
- reworked URL parsing again as www.google.de and google.de both are URLs in the view of browsers
- refactoring for methods calling now async openSource

While I tested what I did, please test as well, if changes you made are still reflected in what I've done.

Also see my comment about a bug in SR5Item.setSource

taMiF commented on src/module/utils/links.ts:
```diff
@@ -1,29 +1,82 @@
+        const parts = candidate.split('.');
+        if (parts.length < 2) return false;
+        const idPart = parts.pop();
+        if (idPart?.length !== 16) return false;
```
This assumption is based on uuid always going Document.Id even when nested. And as far as I know, Foundry uses 16 digit ids. This might be prone breaking, should Foundry change that. I'd prefer a Foundry internal way, though I haven't found one (also haven't looked all that long).

Thogrim1984 replied:
Thats the Foundry Way-to-go.

Thogrim1984 commented:
I wrote the drop function for skills because I had overlooked the fact that they need it separately.

That's why I moved the parseDropData function to module/utils/sheets.ts and refactored other functions that accessed it.

The isUuid check now runs with Foundry's own means, as requested here: https://discord.com/channels/170995199584108546/670336275496042502/1287839670771777620

Unless I have overlooked something, everything should work now.
